src/slider/slider.js

```diff
@@ -0,0 +1,133 @@
+angular.module('ui.bootstrap.slider', ['ui.bootstrap.position'])
+
+.constant('sliderConfig', {
+  updateOnDrag: false,
+  max: 100,
+  min: 0
+})
+
+.directive('slider', ['$document', '$position', 'sliderConfig', function($document, $position, sliderConfig) {
+    return {
+      restrict: 'EA',
+      replace: true,
+      require: '?ngModel',
+      transclude: true,
+      scope: {
+        transition: '@',
+        barClass: '@',
+        btnClass: '@'
+      },
+      templateUrl: 'template/slider/slider.html',
+      link: function(scope, element, attrs, ngModel) {
+        scope.max = angular.isDefined(attrs.max) ? scope.$parent.$eval(attrs.max) : (sliderConfig.max || 100);
+        scope.min = angular.isDefined(attrs.min) ? scope.$parent.$eval(attrs.min) : (sliderConfig.min || 0);
+        scope.transition = scope.transition || sliderConfig['transition'];
+        scope.updateOnDrag = angular.isDefined(attrs.updateOnDrag) ? scope.$parent.$eval(attrs.updateOnDrag) : (sliderConfig['updateOnDrag'] || false);
+
+        scope.btn = element.find('a');
+        scope.bar = element.children().eq(1).children().eq(0);
+        scope.range = scope.max - scope.min;
+
+        scope.hasAnimation = (scope.transition) && (scope.transition !== 'none');
+
+        scope.setAnimation = function(doanimate) {
+          if(doanimate)
+          {
+            scope.btn.css({'transition-property': 'left'});
+            scope.bar.css({'transition-property': 'width'});
+          } else {
+            scope.btn.css({'transition-property': 'none'});
+            scope.bar.css({'transition-property': 'none'});
+          }
+        };
+
+        scope.init = function() {
+          if(scope.hasAnimation)
+          {
+            scope.btn.css({'transition': 'left ' + scope.transition});
+            scope.bar.css({'transition': 'width ' + scope.transition});
+          }
+          scope.setAnimation(scope.hasAnimation);
+        };
+
+        scope.getPercentage = function(value) {
+          return angular.isNumber(value) ? Math.max(0, Math.min(100, Math.round(100 * (value - scope.min) / scope.range))) : 0;
+        };
+
+        scope.percentageToValue = function(perc) {
+          return Math.round(scope.min + (scope.range * perc / 100));
+        };
+
+        scope.setValid = function(value) {
+          var v = parseInt(value, 10);
+          ngModel.$setValidity('slider', (scope.max >= v) && (v >= scope.min));
+          return v;
+        };
+
+        ngModel.$formatters.push(scope.setValid);
+        ngModel.$parsers.push(scope.setValid);
+
+        scope.update = function() {
+          ngModel.$setViewValue(scope.percentageToValue(scope.percent));
+          scope.$apply();
+        };
+
+        ngModel.$render = function() {
+          scope.percent = scope.getPercentage(ngModel.$viewValue);
```

Comment:
If `ngModel.$setViewValue` is called, the $parsers are run and the model value gets updated - what should be done is the `setValid` parser should get pushed into `ngModel.$parsers` and return the percentage value.